src/android/wifiwizard2/WifiWizard2.java

```diff
@@ -1954,4 +1966,100 @@ public void onAvailable(Network network) {
     }
 
   }
+
+  /*
+  *  Specifier one network to connect wifi providing ssid and password
+  *  Author: Anthony Sychev (hello at dm211 dot com)
+  *
+  *  If you not provide pass or algorithm is not set as WPE|WPA|WPA2|WPA3 is represent as default Open network
+  *
+  *  DOC: https://developer.android.com/reference/android/net/wifi/WifiNetworkSpecifier.Builder
+  */
+  private void specifierConnection(CallbackContext callbackContext, JSONArray data) {
+    Log.d(TAG, "WifiWizard2: 211 - Api >= 29 WiFi connection specifier your api version: "+API_VERSION);
+
+    if (API_VERSION >= 29) {
+      if (!validateData(data)) {
+        callbackContext.error("SPECIFIER_INVALID_DATA");
+        Log.d(TAG, "WifiWizard2: 211 - specifierConnection invalid data.");
+        return;
+      }
+
+      try {
+        String SSID = data.getString(0);
+        String PASS = data.getString(1);
+        String Algorithm = data.getString(2);
+        Boolean isHidden = data.getBoolean(3);
+
+        Log.d(TAG, "WifiWizard2: 211 - data: "+ data);
+        Log.d(TAG, "WifiWizard2: 211 - ssid: "+ SSID);
+        Log.d(TAG, "WifiWizard2: 211 - pass: "+ PASS);
+        Log.d(TAG, "WifiWizard2: 211 - Algorithm: "+ Algorithm);
+        Log.d(TAG, "WifiWizard2: 211 - ishidden: "+ isHidden);
+
+        WifiNetworkSpecifier.Builder builder = new WifiNetworkSpecifier.Builder();
+        builder.setSsid(SSID);
+        
+        if(Algorithm.matches("/WEP|WPA|WPA2/gim") && PASS != '')
```

Comment:
You should use `PASS.length() > 0` to check for empty string instead of `PASS != ''`.

```suggestion
        if(Algorithm.matches("/WEP|WPA|WPA2/gim") && PASS.length() > 0)
```

(Same for line 2008)

Reply:
> `PASS != ''` might cause `Empty character literal error` or it might not like comparing string to a (empty) character `''`
> You should use `PASS.length() > 0` to check PASS is not empty string.
> 
> (Same for line 2008)

This needs to be done also for line 2008!